Dockerfile

```diff
@@ -22,10 +20,12 @@ ADD requirements.txt /code/requirements.txt
 ADD package.json /code/package.json
 ADD app/assets /code/app/assets
 
-RUN pip3 install -U -I -r /code/requirements.txt
+RUN git clone git://github.com/yyuu/pyenv.git .pyenv
+ENV PYENV_ROOT /code/.pyenv
+ENV PATH $PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
 
 ADD . /code
 
-RUN yarn compile
+RUN pyenv install &&  pyenv rehash && pip install -U pip setuptools wheel
 
-ENTRYPOINT python3 application.py runserver
+ENTRYPOINT pip install -r requirements.txt && yarn compile && python application.py runserver
```

Comment:
This is kind of an anti-pattern for docker in that the images should be built with all the dependencies and just run. I guess the question is if this image is intended for development or for allowing other people to easily run the system.
The other issue this brings is that failures in the image happen at runtime not compile time as we won't know the image doesn't work until its run.
eg. this image currently fails for me with `ImportError: No module named 'cryptography'` and as our docker compose specifies `restart: always` then it loops around re-attempting infinatly.